### PR TITLE
 1/Completed task and passed test. User Service.

### PR DESCRIPTION
**Pull request by aksoli666**
created model: User;
 created DAO: UserDao;
 created service: UserService with methods add and findByEmail;
                  AuthenticationService with methods login and register;
created own checked RegistrationException and AuthenticationException to use in register() and login() methods respectively; used salt and password-hashing;
used Injector;
used annotation;

### CODE REVIEW

File: src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @Inject
+    private UserDao userDao;
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.findByEmail(email);
+        if (userFromDb.isEmpty()) {
+            throw new AuthenticationException("Can`t login a user by empty email");
```
Comment (MentorMA) [inline src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java:25]:
```suggestion
            throw new AuthenticationException("Can`t login a user by email");
```

Comment (MentorMA) [reply src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java:25]: How do you check that email is empty?

File: src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        User user = userFromDb.get();
+        String hashPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Can`t login a user by email(" + email + ")."
+                + " Wrong password.");
```
Comment (MentorMA) [inline src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java:33]: use formatting for exception if you want add email value and don`t write that only password is wrong!

**Review by MentorMA — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/model/User.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String email;
```
Comment (Rommelua) [inline src/main/java/mate/academy/model/User.java:15]: image

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request closed without merging**